graylog2-web-interface/src/views/components/contexts/WidgetFocusProvider.tsx

```diff
@@ -101,12 +107,26 @@ const useSyncStateWithQueryParams = ({ focusedWidget, focusUriParams, setFocused
       }
 
       setFocusedWidget(nextFocusedWidget);
-      const filter = nextFocusedWidget?.id ? [nextFocusedWidget.id] : undefined;
-      dispatch(setWidgetsToSearch(filter));
+      const searchTypeIds = widgetMapping.get(nextFocusedWidget.id);
 
-      dispatch(execute());
+      dispatch(setSearchTypesToSearch(searchTypeIds?.toArray()));
+      dispatch(executeSearch());
     }
-  }, [focusedWidget, setFocusedWidget, widgetIds, focusUriParams, dispatch]);
+  }, [focusedWidget, setFocusedWidget, widgetIds, focusUriParams, dispatch, widgetMapping]);
+
+  useEffect(() => {
+    if (focusedWidget) {
+      const searchTypeIds = widgetMapping.get(focusedWidget.id)?.toArray() ?? [];
+      const searchTypesToSearchIsUpToDate =
+        searchTypeIds.length === 0 ||
+        !searchTypesToSearch ||
+        searchTypesToSearch.length === 0 ||
+        searchTypeIds.every((id) => searchTypesToSearch?.includes(id));
+      if (!searchTypesToSearchIsUpToDate) {
+        dispatch(setSearchTypesToSearch(searchTypeIds));
+      }
+    }
+  }, [dispatch, focusedWidget, searchTypesToSearch, widgetMapping]);
 };
```

Comment:
This is one of the changes which has a risk of introducing errors. The challenge is to update `searchTypesToSearch` when the search type ids change, while a widget is focused.